src/lib/tools/github.ts

```diff
@@ -15,7 +16,8 @@ export const getPullRequestsTool = tool({
       .describe("The number of the pull request to get the details for"),
   }),
   execute: async ({ repo, owner, pull_number }) => {
-    const octokit = createOctokit();
+    const token = await getTokenForRepository(owner, repo);
+    const octokit = createOctokit(token ?? undefined);
```

Comment:
@tembo if getTokenForRepository returns undefined we dont need the ?? undefined here